File: frontend/src/types/salle.ts
export interface Materiel {
  nom: string;
  type: string;
  quantite: number;
  etat: string;
  date_acquisition?: string | null; // "yyyy-MM-ddTHH:mm" (datetime-local)
}

export interface SalleResponse {
  id: number;
  nom: string;
  type: string;
  capacite: number;
  quantite: number;
  etat: string;
  materiels: Materiel[];
}

// POST /salles, PUT /salles/{id} -> SalleRequestDto
export interface SalleRequestPayload {
  nom: string;
  type: string;
  capacite: number;
  quantite: number;
  etat: string;
  date_acquisition?: string | null;
  materiels: Materiel[];
}

File: backend/src/main/java/com/unireserve/entity/Salle.java
package com.unireserve.entity;

import java.time.LocalDateTime;



import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Table;
import lombok.Data;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.PreUpdate;
import java.util.List;
import jakarta.persistence.CascadeType;
@Entity
@Table(name = "salle")
@Data
public class Salle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;

    private String type;

    private int capacite;

    private int quantite;

    private String etat;

    private LocalDateTime date_acquisition;

    @Column(name="created_at", nullable=false, updatable=false)
    private LocalDateTime createdAt;

    @Column(name="updated_at", nullable=false)
    private LocalDateTime updatedAt;


    @jakarta.persistence.PrePersist
    public void prePersist() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(name = "salle_materiel",joinColumns = @JoinColumn(name="id_salle"),inverseJoinColumns = @JoinColumn(name = "id_materiel"))
    private List<Materiel> materiels;
    
}
